src/librustc/ty/mod.rs

```diff
@@ -1843,6 +1843,7 @@ pub struct VariantDef {
     /// Flags of the variant (e.g. is field list non-exhaustive)?
     flags: VariantFlags,
     /// Recovered?
+    // FIXME: Needs proper doc. Recovered whom from what?
```

Comment:
From some syntactic error.
So e.g. the field list may be incomplete, and we may want to suppress some errors related to that.

Reply:
Introduced in https://github.com/rust-lang/rust/pull/59266.
The recovery terminology is so ubiquitous in the frontend that we assumed it's obvious.

Reply:
So what would be an appropriate doc comment?
```
/// Variant is obtained as part of recovering from a syntactic error.
/// May be incomplete or bogus.
```

Reply:
> The recovery terminology is so ubiquitous in the frontend that we assumed it's obvious.

`Ty` is shared with the middle-end though, and as someone who did a lot of work on the MIR level I have no idea about recovery. :)